front/src/components/surveyExport.js

```diff
@@ -0,0 +1,118 @@
+import axios from 'axios'
+import React, {Component} from "react";
+import {Button} from "reactstrap";
+import {Link} from "react-router-dom";
+
+class SurveyExport extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            survey: -1,
+            frontUrl: "http://localhost:3000",
+            serverUrl: "http://localhost:8080"
+        }
+    }
+
+    componentDidMount() {
+        let survey_id = window.location.href
+        survey_id = survey_id.replace(this.state.frontUrl + "/surveyexport/", "")
```

Comment:
- In the first line the variable is called `survey_id,` but actually the whole url is assigned.
- You can use `this.props.match.params.id` to extract id.